### DISCUSSION TITLE
Nested Array Fields getting the total

### DISCUSSION THREAD

**Discussion by oliseh6123** [react-hook-form/react-hook-form#2977] [Q&A]
HI, Guys. Hi @bluebill1049 blue i have an issue after trying to enhance on your nested Field Array. I am using useWatch to try get the total sum of the nested array field. here is the sandbox https://codesandbox.io/s/quiet-forest-pmykg?file=/src/fieldArray.jsx

I want to indulge your help. i am trying to watch this name={test[${nestIndex}].nestedArray[${k}].field1} from in side the first array form but i am having a repeated total sum.

**Comment by oliseh6123** [comment]
the fields  i want to calculate are nested in an array and also nested in another array where i want to show the total, the sandbox above shows the problem i am having. When i get the total sum for the first set of nested arrays. The total repeats itself when i try to create another set of nested arrays

**Reply by bluebill1049** [reply]
give me sometimes, i will do a demo with number adding in field array.

**Reply by bluebill1049** [reply]
I would try something like the following: 

https://codesandbox.io/s/watchusewatch-calc-tckxm

**Reply by oliseh6123** [reply]
yeah like i have here a nested array fields inside another array field https://codesandbox.io/s/quiet-forest-pmykg like a sample

**Reply by oliseh6123** [reply]
Thanks

**Reply by bluebill1049** [reply]
Take a look at this example: https://codesandbox.io/s/watchusewatch-calc-tckxm i spend a bit more time on it. it's works,

**Reply by oliseh6123** [reply]
@bluebill1049 thanks for taking your time on this but this is not what i am trying to achieve. they should be a parent total for the each of the appended child array like. Like each time i click the append parent button it should produce an append child button and a total field.
so the append[0] child fields will be summed up under parent[0] total field and the append[1] child fields will be summed up under parent[1] total field. Please i hope you understand what i am trying to say thanks.
......
append[0]child[0] field = 5
append[0]child[1] field = 3
parent[0]total field= 8

append[1]child[0] field = 11
append[1]child[1] field = 5
parent[1]total field= 16
